paymentsheet/src/main/java/com/stripe/android/paymentsheet/flowcontroller/DefaultFlowController.kt

```diff
@@ -289,35 +293,46 @@ internal class DefaultFlowController @Inject internal constructor(
         paymentSelection: PaymentSelection?,
         state: PaymentSheetState.Full,
     ) {
-        val mode = viewModel.initializationMode ?: return
+        lifecycleScope.launch {
+            val mode = viewModel.initializationMode ?: return@launch
 
-        val clientSecret = when (mode) {
-            is PaymentSheet.InitializationMode.PaymentIntent -> {
-                PaymentIntentClientSecret(mode.clientSecret)
-            }
-            is PaymentSheet.InitializationMode.SetupIntent -> {
-                SetupIntentClientSecret(mode.clientSecret)
-            }
-            is PaymentSheet.InitializationMode.DeferredIntent -> {
-                TODO("Not implemented yet")
+            val clientSecret = when (mode) {
```

Comment:
Same here: If we move this into a separate class, we can reuse the logic from the complete flow.